service/src/main/java/org/jboss/sbomer/service/feature/sbom/model/RequestEvent.java

```diff
@@ -147,4 +149,35 @@ public RequestEvent save() {
         return this;
     }
 
+    @Override
+    public final boolean equals(Object o) {
```

Comment:
@vibe13 I saw @dwalluck mentioning https://jpa-buddy.com/blog/lombok-and-jpa-what-may-go-wrong/, but can you maybe review this part of the change since you have more experience with it?

Reply:
@dwalluck Since we are changing the equals() to be based on the `id`, I think we should also make the hashcode() rely on the `id` only instead of the hashcode on the whole class?

Reply:
ID and hashcode must be consistent over time and persistence state.
Having equals on ID and static hashcode is universal way to achive that.

Since we have IDs always set before persisting to DB, we can also use the ID for hashcode.